### DISCUSSION TITLE
G7 expiry date on xdrip sync followers not updated

### DISCUSSION THREAD

**Discussion by wirbel-at-vdr-portal** [NightscoutFoundation/xDrip#3594] [Bug reports]
xdrip sync master (xiaomi Note 7)            - nightly 2024.07.23
xdrip sync followers (xiaomi 11T / 11T Pro)  - nightly 2024.07.23

sync works fine, but sensor expiry date is not updated.
Sensor was replaced ~2.5 days ago, so about 8 days are remaining.

system status page on master:
Sensor Status: OK, 2.6 days (correct)

but..

followers (below plot)
Ends Mi., 10:15 PM << wrong. expiry date of previous sensor

expected:
10.5 days - 2.6 days = 7.9 days left
expiry = Sa 11 AM + 7.9 days = Sunday 8 AM

I think the expiry date is just not updated at all on the followers.

**Comment by Navid200** [comment]
Can you show where you are seeing the sensor expiry on the follower?
You should always log into GitHub and look at the discussion and the post in a browser to make sure you see what you have intended to attach.  If you don't see it, we won't see it.

**Comment by wirbel-at-vdr-portal** [comment]
Hi Navid200,

thank you for your response. I'm usually logged in..

Please find the expiry date in the lower middle (below plot) of the attached screenshot from follower. Wednesday relates actually to the past..
Everything else works properly, so this is from my point of view a low prio issue.

Sorry for into German translated string, i would help to translate if there are any questions left.

thanks,
Winfried

![Screenshot_2024-07-27-16-17-50-414_com eveningoutpost dexdrip 1](https://github.com/user-attachments/assets/a51799fa-430f-4862-b4f7-8c609d525824)

**Reply by Navid200** [reply]
OK,
I still cannot see a problem.  

Please see the following showing what I have.  It shows both master and follower, both running July 23, 2024, both showing the same expiry.  
![1](https://github.com/user-attachments/assets/595f9d61-9e52-4242-a9eb-ccd12e859840)  
  
Please note that as I have explained already in another post ([here](https://github.com/NightscoutFoundation/xDrip/discussions/3595)) we are trying to fix xDrip sync.  There may be some side effects and we need everyone to be patient until it becomes fully operational before we can spend any time on tweaking side effects.  

But, I would like to make sure you are using xDrip correctly and have reasonable expectations.  If you can show exactly what the problem is, it may help clarify that one way or another.

**Reply by Navid200** [reply]
Are you sure both master and follower are updated?

**Comment by wirbel-at-vdr-portal** [comment]
>> Are you sure both master and follower are updated?

Absolutely, to 100% - see screenshot. Not even any unexpected interruptions at all. Smoothly. (which is btw great work)

![Image1](https://github.com/user-attachments/assets/a2549c48-c52d-43e7-9dc4-3d0cc5d41c21)

**Comment by jamorham** [comment]
Thanks for reporting. I believe I know what is causing this and will correct it in the next release.

**Comment by wirbel-at-vdr-portal** [comment]
>> OK, I still cannot see a problem.

Really? If so, then pls re-read again carefully.

The end time of this sensor is announced at the follower as **wednesday, 10:15 PM** on the follower, German: "Mi., 10:15 PM". But - there *should* be a time on sunday, the 4th of august early about 8 AM. The time written is the expiry time of the _previous_ sensor.

**Reply by Navid200** [reply]
It was already stated that it will be fixed and you were thanked for reporting it.

**Reply by wirbel-at-vdr-portal** [reply]
> It was already stated that it will be fixed and you were thanked for reporting it.

Sorry, it was too slow to response.

**Comment by wirbel-at-vdr-portal** [comment]
Thanks @jamorham for your time,

it's not urgent, take your time.

If i can help with anything to this project, doing docs, supporting something, few bucks, (..) - Let me know.

**Comment by wirbel-at-vdr-portal** [comment]
Problem is still present in 2024.08.07 nightly.

**Comment by wirbel-at-vdr-portal** [comment]
Date was not updated last 4 or 5 sensor sessions.

**Comment by wirbel-at-vdr-portal** [comment]
It is working again with 2024-09-15. Closing.
